schedule/models/rules.py

```diff
@@ -12,76 +12,94 @@
+        'DAILY': DAILY,
+        'MONTHLY': MONTHLY,
+        'WEEKLY': WEEKLY,
+        'YEARLY': YEARLY,
+        'HOURLY': HOURLY,
+        'MINUTELY': MINUTELY,
+        'SECONDLY': SECONDLY
+    }
+    return compatibiliy_dict[self.frequency]
+
+  def get_params(self):
     """
-    This defines a rule by which an event will recur.  This is defined by the
-    rrule in the dateutil documentation.
-
-    * name - the human friendly name of this kind of recursion.
-    * description - a short description describing this type of recursion.
-    * frequency - the base recurrence period
-    * param - extra params required to define this type of recursion. The params
-      should follow this format:
-
-        param = [rruleparam:value;]*
-        rruleparam = see list below
-        value = int[,int]*
-
-      The options are: (documentation for these can be found at
-      http://labix.org/python-dateutil#head-470fa22b2db72000d7abe698a5783a46b0731b57)
-        ** count
-        ** bysetpos
-        ** bymonth
-        ** bymonthday
-        ** byyearday
-        ** byweekno
-        ** byweekday
-        ** byhour
-        ** byminute
-        ** bysecond
-        ** byeaster
+    >>> rule = Rule(params = "count:1;bysecond:1;byminute:1,2,4,5")
+    >>> rule.get_params()
+    {'count': 1, 'byminute': [1, 2, 4, 5], 'bysecond': 1}
     """
-    name = models.CharField(_("name"), max_length=32)
-    description = models.TextField(_("description"))
-    frequency = models.CharField(_("frequency"), choices=freqs, max_length=10)
-    params = models.TextField(_("params"), null=True, blank=True)
-
-    class Meta:
-        verbose_name = _('rule')
-        verbose_name_plural = _('rules')
-        app_label = 'schedule'
-
-    def rrule_frequency(self):
-        compatibiliy_dict = {
-                'DAILY': DAILY,
-                'MONTHLY': MONTHLY,
-                'WEEKLY': WEEKLY,
-                'YEARLY': YEARLY,
-                'HOURLY': HOURLY,
-                'MINUTELY': MINUTELY,
-                'SECONDLY': SECONDLY
-                }
-        return compatibiliy_dict[self.frequency]
-
-    def get_params(self):
-        """
-        >>> rule = Rule(params = "count:1;bysecond:1;byminute:1,2,4,5")
-        >>> rule.get_params()
-        {'count': 1, 'byminute': [1, 2, 4, 5], 'bysecond': 1}
-        """
-        if self.params is None:
-            return {}
-        params = self.params.split(';')
-        param_dict = []
-        for param in params:
-            param = param.split(':')
-            if len(param) == 2:
-                param = (str(param[0]), [int(p) for p in param[1].split(',')])
-                if len(param[1]) == 1:
-                    param = (param[0], param[1][0])
-                param_dict.append(param)
-        return dict(param_dict)
-
-    def __unicode__(self):
-        """Human readable string for Rule"""
-        return 'Rule %s params %s' % (self.name, self.params)
+    if self.params is None:
```

Comment:
This looks pretty good to me. My +1; wait for Jeremy's +1 perhaps.

On Tue, Mar 3, 2015 at 9:49 AM, sganeriwal <notifications@github.com> wrote:

> In schedule/models/rules.py
> <https://github.com/AltSchool/django-scheduler/pull/2#discussion_r25708859>
> :
>
> > -            return {}
> > -        params = self.params.split(';')
> > -        param_dict = []
> > -        for param in params:
> > -            param = param.split(':')
> > -            if len(param) == 2:
> > -                param = (str(param[0]), [int(p) for p in param[1].split(',')])
> > -                if len(param[1]) == 1:
> > -                    param = (param[0], param[1][0])
> > -                param_dict.append(param)
> > -        return dict(param_dict)
> > -
> > -    def __unicode__(self):
> > -        """Human readable string for Rule"""
> > -        return 'Rule %s params %s' % (self.name, self.params)
> > +    if self.params is None:
>
> Freaking github is horrible. There are minimal changes to the file and it
> is showing all types of false negatives.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/AltSchool/django-scheduler/pull/2/files#r25708859>.
>